external-crates/move/crates/move-analyzer/trace-adapter/src/adapter.ts

```diff
@@ -300,21 +343,28 @@ export class MoveDebugSession extends LoggingDebugSession {
                 value: '(' + value.length + ')[...]',
                 variablesReference: compoundValueReference
             };
-        } else {
+        } else if ('fields' in value) {
```

Comment:
We need to know which "type" of runtime value (`RuntimeValueType`) we are dealing with. Unfortunately, TypeScript-level types are erased at runtime so the way to do it is to check if a given (unique) property is part of a given value. Here's `RuntimeValueType` definition:
```typescript
export type RuntimeValueType = string | CompoundType | IRuntimeRefValue;
export type CompoundType = RuntimeValueType[] | IRuntimeCompundValue;
```
of, if you "unroll" it:
```typescript
export type RuntimeValueType = string | RuntimeValueType[] | IRuntimeCompundValue | IRuntimeRefValue;
```

- `typeof value === 'string'` handles the `string` case,
- `Array.isArray(value)` handles the `RuntimeValueType[]` case
- `'fields' in value` handles the `IRuntimeCompoundValue` case
- `else` handles `IRuntimeRefValue` case

This works because only `IRuntimeCompoundValue` has the `fields` property:

```typescript
export interface IRuntimeCompundValue {
    fields: [string, RuntimeValueType][];
    type: string;
    variantName?: string;
    variantTag?: number;
}
```